src/components/Forms/Artist/index.tsx

```diff
@@ -14,9 +20,23 @@ export default function ArtistForm() {
         handleDeletePhoto,
         artistMutationIsSuccess,
         editArtistMutationIsSuccess,
-        error
+        error,
+        reset,
+        getValues
     } = useArtistForm()
 
+    useEffect(() => {
+        if (data && JSON.stringify(data) !== JSON.stringify(getValues)) {
```

Comment:
We shouldn't need to do this specific of an assertion here. Just check if (data).. Also pass `param` to the `enabled` option on the query.

Reply:
You can pass something like `Boolean(param)` to the options argument like so 

```ts
const { data } = api.artist.get.useQuery({ id: param }, { enabled: Boolean(param) })
```
What this will do is ensure the query won't run if param isn't present.